### DISCUSSION TITLE
Per-project integration should beat user-wide integration

### DISCUSSION THREAD

**Discussion by ScottLangham** [microsoft/vcpkg#27315] [Q&A]
I have a project that uses per-project integration; the `.props` and `.targets` from vcpkg are imported. This worked great until a developer already had vcpkg installed in a directory outside of the project repo which they had used `vcpkg integrate install` with.

`vcpkg integrate install` appears to cause `vcpkg.user.props` to be generated under the users AppData folder, and that imports props from their user-wide vcpkg unconditionally. That results in many build warnings like this:

```
vcpkg.user.props(3,3): warning MSB4011: "D:\vcpkg\scripts\buildsystems\msbuild\vcpkg.props" cannot be imported again. It was already imported at "D:\MyApp\Directory.Build.props". This is most likely a build authoring error. This subsequent import will be ignored.
```

Possibly the user-wide import should have a guard  `Condition="'$(VcpkgPropsImported)'!='true'"` on the Import?  That way the per-project integration can win without generating build warnings.

**Comment by JackBoosY** [comment]
cc @BillyONeal
